proto/spire/service/common/config/v1/config.proto

```diff
@@ -25,6 +29,22 @@ message ConfigureRequest {
 message ConfigureResponse {
 }
 
+message ValidateRequest {
+    // Required. Core SPIRE configuration.
+    CoreConfiguration core_configuration = 1;
+
+    // Required. HCL encoded plugin configuration.
+    string hcl_configuration = 2;
+}
+
+message ValidateResponse {
+    // Required. True when the plugin deems the configuration usable.
+    bool valid = 1;
+
+    // Required. Text message suitable for presentation to end user.
+    string text_message = 2;
```

Comment:
I think we might get some more flexibility over presentation if this is a repeated field where each item represents a distinct, unitary note related to the validity or invalidity:
```suggestion
    repeated string notes = 2;
```

Reply:
Agreed.  It would permit one run to report back two or more errors or advisories, instead of running the `validate` command multiple times to receive the next "hidden by previous issues" error.

I'll make it clear that each message must stand on its own, and that the `notes` field cannot be used as a formatting mechanism for determining presentation oriented line breaks.